parquet-hadoop/src/test/java/org/apache/parquet/filter2/dictionarylevel/DictionaryFilterTest.java

```diff
@@ -464,6 +508,50 @@ public void testGtEqMissingColumn() throws Exception {
         canDrop(gtEq(b, Binary.fromString("any")), ccmd, dictionaries));
   }
 
+  @Test
+  public void testUdpMissingColumn() throws Exception {
+    InInt32UDP udp = new InInt32UDP(ImmutableSet.of(42));
+    IntColumn fake = intColumn("missing_column");
+
+    assertFalse("Should not drop block for any null columns",
+      canDrop(FilterApi.userDefined(fake, udp), ccmd, dictionaries));
+  }
+
+
+  @Test
+  public void testInverseUdpMissingColumn() throws Exception {
+    InInt32UDP udp = new InInt32UDP(ImmutableSet.of(42));
+    IntColumn fake = intColumn("missing_column");
+
+    assertFalse("Should not drop block for any null columns",
+      canDrop(LogicalInverseRewriter.rewrite(FilterApi.userDefined(fake, udp)), ccmd, dictionaries));
+  }
+
+
+  private static final class InInt32UDP extends UserDefinedPredicate<Integer> implements Serializable {
+
+    private final Set<Integer> ints;
+
+    InInt32UDP(Set<Integer> ints) {
+      this.ints = ints;
+    }
+
+    @Override
+    public boolean keep(Integer value) {
+      return ints.contains(value);
+    }
+
+    @Override
+    public boolean canDrop(Statistics<Integer> statistics) {
+      return false;
```

Comment:
is this a stub value?  We ought to be able to check the stats against the integer set to more intelligently filter the file out I'd think?